api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCcrCohortQueries.java

```diff
@@ -0,0 +1,3014 @@
+   * período será considerada a informação registada na primeira ficha.
+   *
+   * @return {@link CohortDefinition}
+   */
+  public CohortDefinition getChildrenWhoReceivedCsbOrNutritionalSuplement() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.setName("Crianças que receberam CSB/suplemento nutricional");
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Health Facility", Location.class));
+
+    cd.addSearch("firstConsultation", map(getPatients1stConsultation(), mapping));
+    cd.addSearch("receivedCsb", map(getChildrenWhoReceivedCsb(), mapping));
+
+    cd.setCompositionString("firstConsultation AND receivedCsb");
+    return cd;
+  }
+
+  public CohortDefinition getInfantAge(boolean greaterThan, Integer Age) {
+    SqlCohortDefinition sqlCohortDefinition = new SqlCohortDefinition();
+    sqlCohortDefinition.setName("Infant Age");
+    sqlCohortDefinition.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("endDate", "End Date", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("location", "Location", Location.class));
+
+    Map<String, Integer> map = new HashMap<>();
+    map.put("92", hivMetadata.getCCRResumoEncounterType().getEncounterTypeId());
+    map.put("Age", Age);
+
+    String query =
+        "SELECT "
+            + "    pr.person_id "
+            + "FROM "
+            + "    person pr "
+            + "        INNER JOIN ( "
+            + "        SELECT "
+            + "            p.patient_id, "
+            + "            MIN(e.encounter_datetime) AS enrollment_date "
+            + "        FROM "
+            + "            patient p "
+            + "                INNER JOIN encounter e "
+            + "                           ON p.patient_id = e.patient_id "
+            + "        WHERE "
+            + "            p.voided = 0 "
+            + "          AND e.voided = 0 "
+            + "          AND e.encounter_type = ${92} "
+            + "          AND e.location_id = :location "
+            + "          AND e.encounter_datetime >= :startDate "
+            + "          AND e.encounter_datetime <= :endDate "
+            + "        GROUP BY "
+            + "            p.patient_id "
+            + "    ) ccr "
+            + "                   ON pr.person_id = ccr.patient_id "
+            + "WHERE "
+            + "    pr.birthdate IS NOT NULL "
+            + "  AND ccr.enrollment_date IS NOT NULL ";
+    if (greaterThan) {
+      query = query + "  AND TIMESTAMPDIFF(MONTH , pr.birthdate, ccr.enrollment_date) >= ${Age}";
+    } else {
+      query = query + "  AND TIMESTAMPDIFF(MONTH , pr.birthdate, ccr.enrollment_date) < ${Age}";
+    }
+
```

Comment:
Please apply a more semantic form to perform this concat operations

Reply:
Resolved